docs/UserGuide.md

```diff
@@ -111,6 +114,12 @@ Adds a company to the address book.
 
 Format: `add company n/NAME a/ADDRESS b/BILLING_DATE p/PHONE`
 
+* The name of the company is case-insensitive when used to identify duplicates 
+and to associate jobs to companies.
+* The billing date is a number from 1 to 28 that signifies the day of the month
+(with reference to the start of the month) that a company will pay their
+employees' salaries.
+
```

Comment:
I think you can consider adding a note here mentioning end of the month option as a future enhancement, just so it is clear. 

We also need to update the Future enhancement on this as well.